src/patternfly/components/TreeView/tree-view.scss

```diff
@@ -137,22 +136,32 @@ $pf-c-tree-view--MaxNesting: 10;
 
 .pf-c-tree-view__content {
   display: flex;
+}
 
-  &:hover {
-    background-color: var(--pf-c-tree-view__node--hover--BackgroundColor);
-  }
+.pf-c-tree-view__content:hover,
+.pf-c-tree-view__content:focus-within {
+  background-color: var(--pf-c-tree-view__node--hover--BackgroundColor);
 }
 
 .pf-c-tree-view__action {
   padding-right: var(--pf-c-tree-view__action--PaddingRight);
   padding-left: var(--pf-c-tree-view__action--PaddingLeft);
   margin-left: auto;
-  color: var(--pf-c-tree-view__action--Color);
   border: 0;
 
+  &:hover {
+    --pf-c-tree-view__action-item--Color: var(--pf-c-tree-view__action-item--hover--Color);
+  }
+
   &:focus {
+    --pf-c-tree-view__action-item--Color: var(--pf-c-tree-view__action-item--focus--Color);
+
     background-color: var(--pf-c-tree-view__action--focus--BackgroundColor);
   }
+
+  .pf-c-tree-view__action-item {
```

Comment:
is there a reason this should be `-item` and not `-icon`, like you did with the toggle/toggle-icon and data-list draggable-button/draggable-button-icon? Same comment here about `color` belonging on the button, and the `-item/icon` class not necessary but we can keep it. It's useful in the toggle-icon because we can apply a transform to that.